autoload/codefmt/zprint.vim

```diff
@@ -0,0 +1,82 @@
+" Copyright 2019 Google Inc. All rights reserved.
+"
+" Licensed under the Apache License, Version 2.0 (the "License");
+" you may not use this file except in compliance with the License.
+" You may obtain a copy of the License at
+"
+"     http://www.apache.org/licenses/LICENSE-2.0
+"
+" Unless required by applicable law or agreed to in writing, software
+" distributed under the License is distributed on an "AS IS" BASIS,
+" WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+" See the License for the specific language governing permissions and
+" limitations under the License.
+
+
+let s:plugin = maktaba#plugin#Get('codefmt')
+
+let s:supported_filetypes = ['clojure']
+
+
+""
+" @private
+" Formatter: zprint
+function! codefmt#zprint#GetFormatter() abort
+  let l:formatter = {
+        \ 'name': 'zprint',
+        \ 'setup_instructions':
+        \ 'Install zprint filter (https://github.com/kkinnear/zprint) ' .
+        \ 'and configure the zprint_executable flag'}
+
+  function l:formatter.IsAvailable() abort
+    return executable(s:plugin.Flag('zprint_executable'))
+  endfunction
+
+  function l:formatter.AppliesToBuffer() abort
+    return index(s:supported_filetypes, &filetype) >= 0
+  endfunction
+
+  ""
+  " Reformat the current buffer with zprint or the binary named in
+  " @flag(zprint_executable), only targeting the range between {startline} and
+  " {endline}.
+  function l:formatter.FormatRange(startline, endline) abort
+    " Must be upper-cased to call as a function
+    let l:ZprintOptions = s:plugin.Flag('zprint_options')
+    if type(l:ZprintOptions) is# type([])
+      " Assign upper-case to lower-case
+      let l:zprint_options = l:ZprintOptions
+    elseif maktaba#value#IsCallable(l:ZprintOptions)
+      " Call upper-case to assign lower-case
+      let l:zprint_options = maktaba#function#Call(l:ZprintOptions)
+    else
+      throw maktaba#error#WrongType(
+            \ 'zprint_options flag must be list or callable. Found %s',
+            \ string(l:ZprintOptions))
+    endif
+    let l:cmd = [s:plugin.Flag('zprint_executable')]
+    call extend(l:cmd, l:zprint_options)
+
+    call maktaba#ensure#IsNumber(a:startline)
+    call maktaba#ensure#IsNumber(a:endline)
+    let l:lines = getline(1, line('$'))
+
+    " zprint doesn't support formatting a range of lines, so format the range
```

Comment:
Can you follow up with zprint to request real range formatting and post a pointer to an issue in their tracker here? For a functional language, hacked range formatting is going to be especially brittle.

Reply:
Will do

Reply:
Done, but I'll also update the PR description to include this link before removing the draft status. https://github.com/kkinnear/zprint/issues/122